packages/ui5-proxy-middleware/src/base/utils.ts

```diff
@@ -195,36 +195,40 @@ export const setHtmlResponse = (res: any, html: string): void => {
  * @param args list of runtime arguments
  * @returns The content of the manifest.json
  */
-export const getManifest = async (args: string[]): Promise<Manifest> => {
+export async function getManifest(args: string[]): Promise<Manifest | undefined> {
     const projectRoot = process.cwd();
     const yamlFileName = getYamlFile(args);
     const ui5YamlPath = join(projectRoot, yamlFileName);
     const webAppFolder = await getWebAppFolderFromYaml(ui5YamlPath);
     const manifestPath = join(projectRoot, webAppFolder, 'manifest.json');
-    const manifest: Manifest = JSON.parse(readFileSync(manifestPath, { encoding: 'utf8' }));
-
-    return manifest;
-};
+    if (existsSync(manifestPath)) {
+        return JSON.parse(readFileSync(manifestPath, { encoding: 'utf8' }));
+    } else {
+        return undefined;
+    }
+}
 
 /**
  * Gets the minUI5Version from the manifest.json.
  *
  * @param args list of runtime args
+ * @param manifest optional already loaded manifest.json
  * @returns The minUI5Version from manifest.json or undefined otherwise
  */
-export async function getUI5VersionFromManifest(args: string[]): Promise<string | undefined> {
-    const manifest = await getManifest(args);
-    return manifest['sap.ui5']?.dependencies?.minUI5Version;
+export async function getUI5VersionFromManifest(args: string[], manifest?: Manifest): Promise<string | undefined> {
+    manifest ??= await getManifest(args);
```

Comment:
We should probably avoid mutating the passed in `manifest` parameter here. It doesn't actually change the `manifest` variable in the calling code and is confusing. If the intent is to fetch the manifest if it's missing, it's better to explicitly return it?

Reply:
Fixed with ea8223bf